error: redeclaration of '_t1'

### Describe the bug

create two files with `const` use tmp vars, it will cause tmp vars' names redeclaration. 

This bug may related to #24517
It may related to commit 05361b75ca50e30016721c0b260c85ecf43dc783

### Reproduction Steps

create a new dir, for example `b`:
create two files: `a.v` and `b.v`
a.v
```v
module main

import json

pub struct JwtHeader {
pub:
        alg string
        typ string
        cty string
}

const f1 = json.encode(JwtHeader{ alg: '1', typ: '2', cty: '3' })
```

b.v
```v
module main

import json

const f2 = json.encode(JwtHeader{ alg: 'a', typ: 'b', cty: 'c' })

fn main() {
        println(f1)
        println(f2)
}
```

run following command :
```
v .
```

### Expected Behavior

create an executable without error.

### Current Behavior

```
$ v .
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/a.01JVM31SPGK53HY6TZTW39027Y.tmp.c:6930: error: redeclaration of '_t1'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

### Possible Solution

When create a `const expr` in v compiler , it should name `_t1`, `_t2`... tmp vars' name globally, and should not reset the `global_tmp_count` between files.

When the v compiler generate `_vinit()`, it combine all `global` consts in one function.


### Additional Information/Context

It may related to commit 05361b75ca50e30016721c0b260c85ecf43dc783

### V version

V 0.4.10 2c42574e6a25b4c5ca8381a63a05a0d0b997436f

### Environment details (OS name and version, etc.)

|V full version      |V 0.4.10 2c42574e6a25b4c5ca8381a63a05a0d0b997436f
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 24.04.2 LTS
|Processor           |8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-9700 CPU @ 3.00GHz
|Memory              |5.19GB/15.51GB
|                    |
|V executable        |/media/HD/github/kbkpbot/v/v
|V last modified time|2025-05-19 10:57:17
|                    |
|V home dir          |OK, value: /media/HD/github/kbkpbot/v
|VMODULES            |OK, value: /home/mars/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /media/HD/github/kbkpbot/v
|                    |
|Git version         |git version 2.43.0
|V git status        |weekly.2025.17-122-g2c42574e
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|gcc version         |gcc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|clang version       |Ubuntu clang version 18.1.3 (1ubuntu1)
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (Ubuntu GLIBC 2.39-0ubuntu8.4) 2.39

--- a/vlib/v/gen/c/consts_and_globals.v
+++ b/vlib/v/gen/c/consts_and_globals.v
@@ -340,10 +340,15 @@ fn (mut g Gen) const_decl_init_later(mod string, name string, expr ast.Expr, typ
 		func := (expr_sym.info as ast.FnType).func
 		def = g.fn_var_signature(func.return_type, func.params.map(it.typ), cname)
 	}
+	init_str := init.str().trim_right('\n')
 	g.global_const_defs[util.no_dots(name)] = GlobalConstDef{
 		mod:       mod
 		def:       '${def}; // inited later'
-		init:      init.str().trim_right('\n')
+		init:      if init_str.count('\n') > 1 {
+			'{\n${init_str}\n}'
+		} else {
+			init_str
+		}
 		dep_names: g.table.dependent_names_in_expr(expr)
 	}
 	if g.is_autofree {

--- a/vlib/v/slow_tests/run_project_folders/issue_24521_constants_in_different_v_files/a.v
+++ b/vlib/v/slow_tests/run_project_folders/issue_24521_constants_in_different_v_files/a.v
@@ -1,0 +1,12 @@
+module main
+
+import json
+
+pub struct JwtHeader {
+pub:
+	alg string
+	typ string
+	cty string
+}
+
+const f1 = json.encode(JwtHeader{ alg: '1', typ: '2', cty: '3' })

--- a/vlib/v/slow_tests/run_project_folders/issue_24521_constants_in_different_v_files/b.v
+++ b/vlib/v/slow_tests/run_project_folders/issue_24521_constants_in_different_v_files/b.v
@@ -1,0 +1,5 @@
+module main
+
+import json
+
+const f2 = json.encode(JwtHeader{ alg: 'a', typ: 'b', cty: 'c' })

--- a/vlib/v/slow_tests/run_project_folders/issue_24521_constants_in_different_v_files/main.v
+++ b/vlib/v/slow_tests/run_project_folders/issue_24521_constants_in_different_v_files/main.v
@@ -1,0 +1,7 @@
+module main
+
+fn main() {
+	println(f1)
+	println(f2)
+	println('OK')
+}

